packages/salesforcedx-vscode-core/test/jest/telemetry/index.test.ts

```diff
@@ -0,0 +1,206 @@
+      await telemetryService.initializeService(
+        mockExtensionContext,
+        'ext_name',
+        'testKey007',
+        'v0.0.1'
+      );
+
+      expect(teleSpy.mock.calls[0]).toEqual([false]);
+    });
+  });
+
+  describe('production mode', () => {
+    beforeEach(() => {
+      mShowInformation = jest
+        .spyOn(window, 'showInformationMessage')
+        .mockResolvedValue(undefined);
+      settings = jest
+        .spyOn(SfdxCoreSettings.prototype, 'getTelemetryEnabled')
+        .mockReturnValue(true);
+      reporter = jest.spyOn(TelemetryReporter.prototype, 'sendTelemetryEvent');
+      exceptionEvent = jest.spyOn(
+        TelemetryReporter.prototype,
+        'sendExceptionEvent'
+      );
+      teleSpy = jest.spyOn(telemetryService, 'setCliTelemetryEnabled');
+      cliSpy = jest
+        .spyOn(telemetryService, 'checkCliTelemetry')
+        .mockResolvedValue(true);
+    });
+
+    afterEach(() => {
+      mShowInformation.mockRestore();
+      settings.mockRestore();
+      reporter.mockRestore();
+      exceptionEvent.mockRestore();
+      teleSpy.mockRestore();
+      cliSpy.mockRestore();
+    });
+
+    it('Should show telemetry info message', async () => {
+      // create vscode extensionContext in which telemetry msg has never been previously shown
+      mockExtensionContext = new MockExtensionContext(
+        false,
+        ExtensionMode.Production
+      );
+
+      await telemetryService.initializeService(
+        mockExtensionContext,
+        'ext_name',
+        'testKey007',
+        'v0.0.1'
+      );
+
+      const telemetryEnabled = await telemetryService.isTelemetryEnabled();
+      expect(telemetryEnabled).toEqual(true);
+
+      showTelemetryMessage(mockExtensionContext);
+      expect(mShowInformation).toHaveBeenCalledTimes(1);
+      expect(teleSpy.mock.calls[0]).toEqual([true]);
+    });
+
+    it('Should not show telemetry info message', async () => {
+      // create vscode extensionContext in which telemetry msg has been previously shown
+      mockExtensionContext = new MockExtensionContext(
+        true,
+        ExtensionMode.Production
+      );
+
+      await telemetryService.initializeService(
+        mockExtensionContext,
+        'ext_name',
+        'testKey007',
+        'v0.0.1'
+      );
+
+      const telemetryEnabled = await telemetryService.isTelemetryEnabled();
+      expect(telemetryEnabled).toEqual(true);
+
+      showTelemetryMessage(mockExtensionContext);
+      expect(mShowInformation).not.toHaveBeenCalled();
+      expect(teleSpy.mock.calls[0]).toEqual([true]);
+    });
+
+    it('Should initialize telemetry reporter and allow undefined aiKey', async () => {
+      // create vscode extensionContext
+      mockExtensionContext = new MockExtensionContext(
+        true,
+        ExtensionMode.Production
+      );
+
+      await telemetryService.initializeService(
+        mockExtensionContext,
+        'ext_name',
+        undefined,
```

Comment:
Oddly, this test is failing on my end due to this exception:`test/jest/telemetry/index.test.ts:196:9 - error TS2345: Argument of type 'undefined' is not assignable to parameter of type 'string'.`. However, the aiKey in this function can be string or undefined. Do you have a clue why it's failing?

Reply:
But, it passed with GitHub actions 🤔

Reply:
It passed in my local. 👍